### DISCUSSION TITLE
How to update table has primary key column and column with list data type? Why UPDATE query return 'Duplicate key' error?

### DISCUSSION THREAD

**Discussion by lpfy** [duckdb/duckdb#10701] [Q&A] [answered]
This behaviour happens for both Python and WASM (https://shell.duckdb.org/) environments. For example:

```
create or replace table test (
	aid INT, 
        comments TEXT,
	alist STRUCT(username VARCHAR, lid INT)[],
	PRIMARY KEY (aid)
);

insert into test select 1, 'abc', [{username:'t1',lid:1},{username:'t2',lid:2}];
insert into test select 2, 'def', [{username:'t3',lid:3},{username:'t4',lid:4}];
UPDATE test SET comments = 'test test' WHERE aid = 1;
UPDATE test SET alist = [{username:'t5',lid:5},{username:'t6',lid:6}] WHERE aid = 1;
```

the last update query will return the following error:
`Constraint Error: Duplicate key "aid: 1" violates primary key constraint. If this is an unexpected constraint violation please double check with the known index limitations section in our documentation (docs - sql - indexes).`

I have read the documents, however, I am not updating the primary key column, the comments column can be updated, just the column with list data type is causing the problem.

**Comment by ccage-simp** [comment]
Experiencing the same. Seems like a bug.

edit: probably related to https://github.com/duckdb/duckdb/discussions/8666...

**Reply by acviana** [reply]
I think this is also related to the discussion in this issue: https://github.com/duckdb/duckdb/issues/11288

If I'm understanding correctly updates are converted deletes and inserts when there is an index or uniqueness constrain (see documentation link in the issue above).

**Reply by lpfy** [reply, marked as answer]
Issue had been resolved, please see #11915, and also the documentation https://duckdb.org/docs/sql/data_types/list
